packages/tools/src/utilities/segmentation/createLabelmapMemo.ts

```diff
@@ -47,6 +44,16 @@ export function restoreMemo(isUndo?: boolean) {
   });
   const slices = useVoxelManager.getArrayOfModifiedSlices();
   triggerSegmentationDataModified(this.segmentationId, slices);
+
+  // Trigger the LABELMAP_UNDO event
+  eventTarget.dispatchEvent(
```

Comment:
Can we add that to the history class instead, and include the memo itself in the detail rather than trying to make it segmentation specific?  It should also say whether it is undo or redo.